test/post.js

```diff
@@ -121,7 +129,7 @@ test('content-type header is not overriden when object in options.body', async t
 	t.is(headers['content-type'], 'doge');
 });
 
-test('throws when json body is not a plain object', async t => {
+test('throws when json body is not a plain object or array', async t => {
```

Comment:
If we do choose to split the argument validation for `json` from `form` we should also have two tests to make sure both work the way we expect.